cpp/test/sg/rproj_test.cu

```diff
@@ -132,7 +135,7 @@ class RPROJTest : public ::testing::Test {
   }
 
   void epsilon_check() {
-    size_t D = johnson_lindenstrauss_min_dim(N, epsilon);
+    int D = johnson_lindenstrauss_min_dim(N, epsilon);
```

Comment:
is this change from `size_t` to `int` intended?

Reply:
It is. The changes I had to make to the test resulted from Thejaswi's addition of the `Index_` template param to the distance prim function, which requires `m`, `n`, and `k` to be of the same type. This was just the path of least resistence and doesn't really effect anything in the tests.